packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -0,0 +1,345 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import { BreadcrumbItem, EditDialog, IFilterApi, ISelectionApi, PrettyBytes, useDataGridRemote } from "@comet/admin";
+import { DataGrid } from "@mui/x-data-grid";
+import * as React from "react";
+import { FileRejection, useDropzone } from "react-dropzone";
+import { FormattedDate, FormattedMessage, FormattedTime, useIntl } from "react-intl";
+import { useDebouncedCallback, useThrottledCallback } from "use-debounce";
+
+import { GQLDamFolderQuery, GQLDamFolderQueryVariables, GQLDamItemsListQuery, GQLDamItemsListQueryVariables } from "../../graphql.generated";
+import { useDamAcceptedMimeTypes } from "../config/useDamAcceptedMimeTypes";
+import { DamConfig, DamFilter } from "../DamTable";
+import AddFolder from "../FolderForm/AddFolder";
+import EditFolder from "../FolderForm/EditFolder";
+import { clearDamItemCache } from "../helpers/clearDamItemCache";
+import { isFile } from "../helpers/isFile";
+import { isFolder } from "../helpers/isFolder";
+import DamContextMenu from "./DamContextMenu";
+import { useFileUpload } from "./fileUpload/useFileUpload";
+import { damFolderQuery, damItemsListQuery } from "./FolderDataGrid.gql";
+import * as sc from "./FolderDataGrid.sc";
+import { FolderHead } from "./FolderHead";
+import { DamSelectionFooter } from "./footer/SelectionFooter";
+import { DamUploadFooter } from "./footer/UploadFooter";
+import { DamItemLabelColumn } from "./label/DamItemLabelColumn";
+import { useDamSearchHighlighting } from "./useDamSearchHighlighting";
+
+export type DamItemSelectionMap = Map<string, "file" | "folder">;
+
+interface FolderDataGridProps extends DamConfig {
+    id?: string;
+    breadcrumbs?: BreadcrumbItem[];
+    filterApi: IFilterApi<DamFilter>;
+    selectionApi: ISelectionApi;
+}
+
+const FolderDataGrid = ({
+    id,
+    filterApi,
+    breadcrumbs,
+    selectionApi,
+    hideContextMenu,
+    hideArchiveFilter,
+    hideMultiselect,
+    renderDamLabel,
+    ...props
+}: FolderDataGridProps): React.ReactElement => {
+    const intl = useIntl();
+    const apolloClient = useApolloClient();
+
+    const [selectionMap, setSelectionMap] = React.useState<DamItemSelectionMap>(new Map());
+    const [showSelectionFooter, setShowSelectionFooter] = React.useState<boolean>(false);
+    const [showUploadFooter, setShowUploadFooter] = React.useState<boolean>(false);
+    const [uploadTargetFolderName, setUploadTargetFolderName] = React.useState<string>();
+
+    const throttledShowSelectionFooter = useThrottledCallback(() => {
```

Comment:
Why do we need throttling?

Reply:
The hover style changes when dragging an item seem a bit less laggy with throttling. But there isn't a huge difference so we could also remove it

Reply:
Does the visibility of the footer change while dragging an item?

Reply:
Yes. Because you can drop on the DataGrid (= current folder) or on a subfolder. 

If onDragLeave is triggered for the DataGrid, it does not know if you are now dragging over a subfolder or if you actually left the drag area. Therefore, in the DataGrid hideFooter() is called and in the subfolder (DamItemLabelColumn) showFooter() is called

Reply:
As far as I understand it, this problem happens by showing/hiding the selection footer when an upload attempt starts. Have you considered these two alternatives that may solve the issue?

1. Clear the selection upon uploading
2. Don't hide the selection footer, but render the upload footer above it?

Furthermore, I believe you may remove both `showXXXFooter` states by deriving the visibility from other states: The upload footer should be shown when the folder to upload to is not undefined (`Boolean(uploadFolderName)`), and the selection footer should be shown when there is a selection (`selectionMap.size > 0`) and no upload (`showUploadFooter === false`). This would probably remove the need for those show/hide functions as well.